eo/notebooks/process_management.ipynb

```diff
@@ -0,0 +1,843 @@
+    "[Docker documentation for python developpement](https://docs.docker.com/language/python/)\n",
+    "\n",
+    "In this case we name our docker image of sentinesat_wrapper : <u>`daccs-eo-sentinelsat` </u>\n",
+    "\n",
+    "<b> Execution example </b></br>  \n",
+    "```shell\n",
+    "docker run daccs-eo-sentinelsat -d --credentials <credentials_file_path> --name <image_name> \n",
+    "```\n",
+    "\n",
+    "As we can see the only thing that changed is that we now execute our code trought docker which contains all the needs for our code to work.\n",
+    "</br></br>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e6f644b7",
+   "metadata": {},
+   "source": [
+    "<b> GitLab Container Registry </b>\n",
+    "</br>\n",
+    "\n",
+    "In our case, after the docker build of the image, the docker container image is then pushed on our [Gitlab Registry](https://docs.gitlab.com/ee/user/packages/container_registry/) so that the image can be pulled as it would from the Docker Hub for example. \n",
+    "The name of the image that we will use in the next steps will now becomes the url pointing to the registry holding our docker container image: \n",
+    "<u>`registry.gitlab.com/crim.ca/public/daccs/daccs/daccs-eo-sentinelsat` </u>\n",
+    "\n",
+    "</br>\n",
+    "The execution command becomes:\n",
+    "\n",
+    "```shell \n",
+    "docker run registry.gitlab.com/crim.ca/public/daccs/daccs/daccs-eo-sentinelsat -d --credentials <credentials_file_path> --name <image_name> \n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "737f6007",
+   "metadata": {},
+   "source": [
+    "### 1.3 CWL document\n",
+    "\n",
+    "We use the application definition configuration provided by [Common Workflow Language](https://www.commonwl.org/user_guide/#common-workflow-language-user-guide) to provide a description of a process to be executed by Weaver. \n",
+    "\n",
+    "A CWL document serves as an interface between our module and all it needs to be executed by Weaver. \n",
+    "\n",
+    "\n",
+    "In this CWL document, there are 3 main part:\n",
+    "1. Docker requirements\n",
+    "    - This is the image name or the url of the image to be pulled from a registry by Weaver\n",
+    "2. Inputs\n",
+    "    - These are the arguments to be passed to our docker to python code \n",
+    "    - In our case : ```-d --credentials <credentials_file_path> --name <image_name>```\n",
+    "3. Ouputs\n",
+    "    - In this case the output is a downloaded file, so we are just going to check if the file exist\n",
+    "\n",
+    "All set inside a CWL yaml file : <b> sentinelsat-application.yml </b>\n",
```

Comment:
"yaml" -> YAML

refer to the "demo/sentinelsat-application.yml" location so it can be found more easily.